datalake/src/main/java/com/sequenceiq/datalake/flow/SdxReactorFlowManager.java

```diff
@@ -55,20 +55,17 @@
     @Inject
     private ThreadBasedUserCrnProvider threadBasedUserCrnProvider;
 
-    @Inject
-    private ThreadBasedRequestIdProvider threadBasedRequestIdProvider;
-
     public void triggerSdxCreation(Long sdxId) {
         String selector = ENV_WAIT_EVENT.event();
         String userId = threadBasedUserCrnProvider.getUserCrn();
-        String requestId = threadBasedRequestIdProvider.getRequestId();
+        String requestId = MDCBuilder.getRequestId().get();
         notify(selector, new SdxEvent(selector, sdxId, userId, requestId));
     }
 
     public void triggerSdxDeletion(Long sdxId, boolean forced) {
         String selector = SDX_DELETE_EVENT.event();
         String userId = threadBasedUserCrnProvider.getUserCrn();
-        String requestId = threadBasedRequestIdProvider.getRequestId();
+        String requestId = MDCBuilder.getRequestId().get();
```

Comment:
Is it a good approach to use the MDCBuilder for getting the request ID here? MDC has the purpose of building the MDC context, I think this is abusing it.

Reply:
ok, i'll create a util class to get it